### DISCUSSION TITLE
"Unrecognized function or variable 'insidepoly_dblengine'"

### DISCUSSION THREAD

**Discussion by M-Eddahbi** [mtex-toolbox/mtex#1756] [Ask Anything]
He Ralf,

I´m working in example given in "Documentation": Topic =EBSD and "KAM", using Mtex 5.8.o version.

-----------------------------------------------------
1. % step

mtexdata ferrite
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));

ebsd(grains(grains.grainSize <= 3)) = [];  % remove one-three pixel grains
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));
grains = smooth(grains,5);

plot(ebsd('indexed'),ebsd('indexed').orientations)
hold on
plot(grains.boundary,'lineWidth',1.5)
hold off

2. % step: gridify for simplicity

ebsd = ebsd.gridify;
kam = ebsd.KAM / degree;

plot(ebsd,kam,'micronbar','off'); % lets plot it
caxis([0,15])
mtexColorbar
mtexColorMap LaboTeX
hold on
plot(grains.boundary,'lineWidth',1.5)
hold off

3. % step: Threshold

plot(ebsd,ebsd.KAM('threshold',2.5*degree) ./ degree,'micronbar','off')
caxis([0,2])
mtexColorbar
mtexColorMap LaboTeX
hold on
plot(grains.boundary,'lineWidth',1.5)
hold off

4. % step: Threshold+filter

F = halfQuadraticFilter; % chose a denoising filter
F.alpha = 0.5;

ebsdS = smooth(ebsd,F,'fill',grains); 
plot(ebsdS,ebsdS.KAM('threshold',2.5*degree) ./ degree,'micronbar','off'); % plot the first order KAM

caxis([0,2])
mtexColorbar
mtexColorMap LaboTeX
hold on
plot(grains.boundary,'lineWidth',1.5)
hold off

Till step 3 is Ok but when executing the step 4 in: **ebsdS = smooth(ebsd,F,'fill',grains)**, the mtex out put these errors:

----------
Unrecognized function or variable 'insidepoly_dblengine'.

Error in insidepoly (line 188)
        [in, on] = insidepoly_dblengine(x, y, Px1, Py1, Px2, Py2, ontol);

Error in grain2d/checkInside (line 91)
    [in,on] = insidepoly(xy(:,1),xy(:,2),V(p,1),V(p,2));

Error in grain2d/checkInside (line 61)
  isInside = grains.checkInside(xy+repmat(uc(1,:),size(xy,1),1),'includeBoundary');

Error in EBSD/fill (line 52)
isInside = checkInside(grains,ebsd.subSet(nanId));

Error in EBSD/smooth (line 57)
if check_option(varargin,'fill') || check_option(varargin,'extrapolate'), ebsd = fill(ebsd,varargin{:}); end
----------------

Please, could you help me to resolve this problem. 

Thank you very much

Eddahbi

**Comment by ralfHielscher** [comment]
the file insidepoly_dblengine.mexw64 should be in the directory
mtex/mex/
could you please check it is there?
This is assuming you have a windows machine. If you have a mac it might be necessary to recompile the mex files by executing 
```mex_install``` within Matlab while being in  that directory.

Ralf.

**Reply by M-Eddahbi** [reply]
He Ralf,

Thank you for your answer.
Yes, I have a windows machine, but the file-- insidepoly_dblengine.mexw64--is not found in the directory mtex/mex/.

Best regards
Eddahbi

**Reply by ralfHielscher** [reply]
Then the archive has not been correctly extracted on your machine. I just checked, the file ist part of the zip file.

Ralf.

**Reply by M-Eddahbi** [reply]
He Ralf,

I download the version 5.8.0 another time but the file still abscent in the extract mex folder. However, I download the version 5.8.2 and such file exists in the directory mex. So, I remplaced the old mex folder by the new one, and now the the problem is resolved.

JUST A QUESTION.

IS THE MTEX-PROGRAM USES THE MATALAB COMPILER (MinGW-64) OR THE ONE LINKED TO WINDOWS?

Thank you for your help

Eddahbi

**Reply by ralfHielscher** [reply]
You should go with MTEX 5.8.2 since it is a pure bug fix release, which e.g. fixes this bug :)

As for your question: I think for the precompolied files we used the one which is default in Matlab.

Ralf.

**Reply by M-Eddahbi** [reply]
Thank you, Ralf.
